src/group_theory/solvable.lean

```diff
@@ -0,0 +1,178 @@
+  exact ⟨p, h₁ hp, q, h₂ hq, rfl⟩,
+end
+
+lemma general_commutator_eq_normal_closure_self (H₁ H₂ : subgroup G) [H₁.normal]
+  [H₂.normal] : general_commutator H₁ H₂ = normal_closure (general_commutator H₁ H₂) :=
+eq.symm normal_closure_eq_self_of_normal
+
+lemma general_commutator_def' (H₁ H₂ : subgroup G) [H₁.normal] [H₂.normal] :
+  general_commutator H₁ H₂ = normal_closure {x | ∃ (p ∈ H₁) (q ∈ H₂), p * q * p⁻¹ * q⁻¹ = x} :=
+by rw [general_commutator_eq_normal_closure_self, general_commutator,
+  normal_closure_closure_eq_normal_closure]
+
+lemma general_commutator_le (H₁ H₂ : subgroup G) (K : subgroup G) :
+  general_commutator H₁ H₂ ≤ K ↔ ∀ (p ∈ H₁) (q ∈ H₂), p * q * p⁻¹ * q⁻¹ ∈ K :=
+begin
+  rw [general_commutator, closure_le],
+  split,
+  { intros h p hp q hq,
+    exact h ⟨p, hp, q, hq, rfl⟩, },
+  { rintros h x ⟨p, hp, q, hq, rfl⟩,
+    exact h p hp q hq, }
+end
+
+lemma general_commutator_le_right (H₁ H₂ : subgroup G) [h : normal H₂] :
+  general_commutator H₁ H₂ ≤ H₂ :=
+begin
+  rw general_commutator_le,
+  intros p hp q hq,
+  exact mul_mem H₂ (h.conj_mem q hq p) (inv_mem H₂ hq),
+end
+
+lemma general_commutator_le_left (H₁ H₂ : subgroup G) [h : normal H₁] :
+  general_commutator H₁ H₂ ≤ H₁ :=
+begin
+  rw general_commutator_le,
+  intros p hp q hq,
+  rw (show p * q * p⁻¹ * q⁻¹ = p * (q * p⁻¹ * q⁻¹), by group),
+  convert mul_mem H₁ hp (h.conj_mem p⁻¹ (inv_mem H₁ hp) q),
+end
+
+@[simp] lemma general_commutator_bot_eq_bot (H : subgroup G) : general_commutator H ⊥ = ⊥ :=
+begin
+  rw eq_bot_iff,
+  exact general_commutator_le_right H ⊥,
+end
+
+@[simp] lemma bot_general_commutator_eq_bot (H : subgroup G) : general_commutator ⊥ H = ⊥ :=
+begin
+  rw eq_bot_iff,
+  exact general_commutator_le_left ⊥ H,
+end
+
+lemma general_commutator_le_inf (H₁ H₂ : subgroup G) [normal H₁] [normal H₂] :
+  general_commutator H₁ H₂ ≤ H₁ ⊓ H₂ :=
+by simp only [general_commutator_le_left, general_commutator_le_right, le_inf_iff, and_self]
+
+end general_commutator
+
+section derived_series
+
+variables (G)
+
+/-- The derived series of the group `G`, obtained by starting from the subgroup `⊤` and repeatedly
+  taking the commutator of the previous subgroup with itself for `n` times. -/
+def derived_series : ℕ → subgroup G
+| 0       := ⊤
+| (n + 1) := general_commutator (derived_series n) (derived_series n)
+
+@[simp] lemma derived_series_zero : derived_series G 0 = ⊤ := rfl
+
+@[simp] lemma derived_series_succ (n : ℕ) :
+  derived_series G (n + 1) = general_commutator (derived_series G n) (derived_series G n) := rfl
+
+lemma derived_series_normal (n : ℕ) : (derived_series G n).normal :=
+begin
+  induction n with n ih,
+  { exact subgroup.top_normal, },
+  { haveI : (derived_series G n).normal := ih,
+    rw derived_series_succ,
+    exact general_commutator_normal (derived_series G n) (derived_series G n), }
+end
+
+lemma commutator_eq_general_commutator_top_top :
+  commutator G = general_commutator (⊤ : subgroup G) (⊤ : subgroup G) :=
```

Comment:
```suggestion
lemma general_commutator_eq_commutator :
  general_commutator (⊤ : subgroup G) (⊤ : subgroup G) = commutator G :=
```
I think this *could* be simp, but I'm fine with being conservative, and leaving it out of the default simp-set.

Reply:
I think making it a `simp` lemma sounds like a good idea.